src/plugins/action_update_review_status/src/updateReviews.ts

```diff
@@ -0,0 +1,70 @@
+import databaseMethods from './databaseMethods'
+
+type ReviewStatus = 'Draft' | 'Submitted' | 'Changes Requested' | 'Pending' | 'Locked'
+interface Review {
+  reviewId: number
+  reviewAssignmentId: number
+  applicationId: number
+  reviewerId: number
+  level: number
+  reviewStatus: ReviewStatus
+}
+
+module.exports['updateReviews'] = async function (input: any, DBConnect: any) {
+  const db = databaseMethods(DBConnect)
+
+  console.log('Updating reviews status...')
+
+  const { applicationId, changedApplicationResponses = [] } = input
+
+  const reviewsToUpdate = []
+
+  console.log('changedApplicationResponses', changedApplicationResponses)
+
+  try {
+    // Get reviews/review assignments (with status) matching application_id
+    const reviews = (await db.getAssociatedReviews(applicationId)).filter((review: Review) =>
+      ['Submitted', 'Locked', 'Draft'].includes(review.reviewStatus)
+    )
+    // Deduce which ones should be updated
+    for (const review of reviews) {
+      const { reviewAssignmentId, level, reviewStatus } = review
+      if (level > 1) reviewsToUpdate.push({ ...review, reviewStatus: 'Pending' })
+      else if (await haveAssignedResponsesChanged(reviewAssignmentId))
+        reviewsToUpdate.push({ ...review, reviewStatus: 'Pending' })
+      else if (reviewStatus === 'Pending' || reviewStatus === 'Locked')
+        reviewsToUpdate.push({ ...review })
+    }
+    console.log('reviewsToUpdate', reviewsToUpdate)
+
+    // Update review statuses
+    for (const review of reviewsToUpdate) {
+      const { reviewId, reviewStatus } = review
+      DBConnect.addNewReviewStatusHistory(reviewId, reviewStatus)
+    }
+
+    return {
+      status: 'Success',
+      error_log: '',
+      output: {
+        updatedReviews: reviewsToUpdate,
+      },
+    }
+  } catch (error) {
+    console.log(error.message)
+    return {
+      status: 'Fail',
+      error_log: 'There was a problem updating review statuses.',
+    }
+  }
+
+  async function haveAssignedResponsesChanged(reviewAssignmentId: number) {
+    const questionAssignments = await db.getReviewQuestionAssignments(reviewAssignmentId)
+    return changedApplicationResponses.reduce(
```

Comment:
some would be better here i think (jut match the first one).

another place where I think postgresql would be useful (this is solid thought dont' need to change).

select count(*) from review_assignment where template_element_id = any ($1) and assignment_id = ($2)

Reply:
> some would be better here i think (jut match the first one).

You saying that `.some()` would stop checking as soon as it matches the first inclusion? Good to know. I was planning on using a traditional `for.. of` loop and using `break` as soon as it gets a match, but I figured the list of items would be so small that it doesn't really matter.

Reply:
yeah, I find it much easier to see then the reduce logic (frankly I find reduce harder to follow then forEach)